main.c

```diff
@@ -0,0 +1,203 @@
+			if (mode == 3) shell_input(optarg, "dictionary");
+			break;
+		case 'i':
+			if (!fileexists(optarg)) {
+				shell_error("Input file path invalid");
+				return EXIT_FAILURE;
+			}
+			mode = 3;
+            strcpy(file_name,optarg);
+			// if file name is not valid, print error
+			break;
+		case 'v':
+			if (!fileexists(optarg)) {
+				shell_error("Input file path invalid");
+				return EXIT_FAILURE;
+			}
+			mode = 2;
+            strcpy(file_name,optarg);
+			break;
+		case 'q':
+			if (!fileexists(optarg)) {
+				shell_error("Input file path invalid");
+				return EXIT_FAILURE;
+			}
+			mode = 1;
+			strcpy(file_name,optarg);
+			break;
+		case 's':
+            if (strstr(optarg,".txt\0") == NULL) {    // does not save to a *.txt file
+                  shell_error("Input save file path invalid");
+                  return EXIT_FAILURE;
+            }
+			strcpy(save_file,optarg);
+            if (mode == 3) shell_input(optarg,"file save destination");
+			break;
+		default:
+			shell_usage();
+            exit(0);
+		}
+	}
+
+	int *quit = malloc(sizeof(int*)); // this enables return to main page, should user choose it at save_page
+	*quit = 0;
+
+  while (!(*quit)) {
+	if (fileexists(file_name)) {	// if file exists, then bypass main page
+		*quit = 1;
+	}
+
+	/* main page: activated if there is no file to be parsed.
+	   can open help page, quit, or load filename / dictname */
+	main_page(quit, &mode, file_name, dict_name);
+	if ((*quit) == 2) { // user selected "quit" in main_page
+		shell_outro();
+		return 0;
+	}
+
+	/*
+		Initialize dictionary, declare names of files to be used
+	*/
+	dict_t* dict = dict_new();
+	if (dict_read(dict, dict_name) == EXIT_FAILURE) {
+		printf("Trouble reading dictionary, exiting program\n");
+        exit(0);
+	}
+
+	/*
+		Starting to Parse file!
+	 */
+	char* md = modename(mode);
+
+    if (mode == 3) {
+
+        printf("\n\n============================================================\n"
+                   "=================== Editing Started With ===================\n"
+                   "============================================================\n\n");
+	    printf("file: %s\n", file_name);
+	    printf("dictionary: %s\n", dict_name);
+	    printf("mode: %s\n\n", md);
+    }
+
+	char** result=NULL;
+	// Execute either interactive or batch mode, and save file at end
+	switch (mode) {
+		case 1: result = batch_mode(file_name, dict, quit, 0); // pass in dictionary
+			break;
+		case 2: result = batch_mode(file_name, dict, quit, 1); // pass in dictionary 
+			break;
+		case 3: result = interactive_mode(file_name, dict, quit); // pass in dictionary
+			break;
+		default:
+			break;
+	}
+
+
+    if (mode != 2 && result != NULL) {	// Save file, a functionality unnecessary for verbose batch mode
+    	if (mode == 3) printf("\n\nSpellcheck successful!\n\n");
+
+    	md = strstr(save_file,".txt");
+    	
+    	if (md != NULL) {
+    		save_corrections(save_file, result);
+    		*quit=1;
```

Comment:
I don't know what's going on here, but this should definitely just be `should_quit = true`. This is a terrible convention and really needs to be avoided. Nobody should have to actually expend effort to figure out how such a simple task is implemented.